app/javascript/components/batch_update/components/BatchUpdateForm.js

```diff
@@ -7,17 +7,21 @@ import { Link } from './Links'
 import { Button } from '@artsy/palette'
 import { colors } from './Layout'
 import GeneInput from './GeneInput'
-import { GeneAutosuggest } from './Autosuggest'
+import { GeneAutosuggest, TagAutosuggest } from './Autosuggest'
 import Overlay from './Overlay'
 import ConfirmationModal from './ConfirmationModal'
 import { submitBatchUpdate } from 'lib/rosalind-api'
+import { SelectedTag } from './Selected'
 
 class BatchUpdateForm extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
       geneValues: {},
       isConfirming: false,
+      existingTags: [],
+      tagsToAdd: [],
+      tagsToRemove: [],
```

Comment:
👍 this makes sense.

Another way to structure this is to treat tag values as a dictionary, maybe we can discuss pros/cons.